python/taulabs/uavtalk.py

```
@@ -1,322 +1,236 @@
+    0x4e, 0x49, 0x40, 0x47, 0x52, 0x55, 0x5c, 0x5b, 0x76, 0x71, 0x78, 0x7f, 0x6a, 0x6d, 0x64, 0x63,
+    0x3e, 0x39, 0x30, 0x37, 0x22, 0x25, 0x2c, 0x2b, 0x06, 0x01, 0x08, 0x0f, 0x1a, 0x1d, 0x14, 0x13,
+    0xae, 0xa9, 0xa0, 0xa7, 0xb2, 0xb5, 0xbc, 0xbb, 0x96, 0x91, 0x98, 0x9f, 0x8a, 0x8d, 0x84, 0x83,
+    0xde, 0xd9, 0xd0, 0xd7, 0xc2, 0xc5, 0xcc, 0xcb, 0xe6, 0xe1, 0xe8, 0xef, 0xfa, 0xfd, 0xf4, 0xf3
+]
+
+def processStream(uavo_defs, useWallTime=False, logTimestamps=False):
+    global wantHint
+
+    # These are used for accounting for timestamp wraparound
+    timestampBase = 0
+    lastTimestamp = 0
+
+    received = 0
+
+    packetBytes=''
+
+    while True:
+        if logTimestamps:
+            while len(packetBytes) < logHeaderFmt.size:
+                rx = yield None
+
+                if rx is None:
+                    return
+
+                packetBytes = packetBytes + rx
+
+            overrideTimestamp, logHdrLen = logHeaderFmt.unpack_from(packetBytes,0)
+
+            packetBytes = packetBytes[logHeaderFmt.size:]
+
+        # Ensure we have enough room for all the
+        # plain required fields to avoid duplicating
+        # this code lots.
+        # sync(1) + type(1) + len(2) + objid(4) 
+
+        while (len(packetBytes) < headerFmt.size) or (packetBytes[0] != chr(SYNC_VAL)):
+            #print "waitingsync len=%d"%(len(packetBytes))
+
+            rx = yield None
+
+            if rx is None:
+                #end of stream, stopiteration
+                return
+
+            packetBytes = packetBytes + rx
+
+            for i in xrange(len(packetBytes)):
+                if packetBytes[i] == chr(SYNC_VAL):
+                    break
+
+            # Trim off irrelevant stuff, loop and try again
+            packetBytes=packetBytes[i:]
+
+        (sync, packetType, packetLen, objId) = headerFmt.unpack_from(packetBytes,0)
+
+        if (packetType & TYPE_MASK) != TYPE_VER:
+            print "badver %x"%(packetType)
```

Basically lets the user know if they've lost sync on some frames and why.  Could throw, but I would rather not fail to read a file if there's a little glitch in the middle.  Could silently ignore.